application/src/main/java/org/togetherjava/tjbot/features/chaptgpt/ChatGptCommand.java

```diff
@@ -1,37 +1,85 @@
 package org.togetherjava.tjbot.features.chaptgpt;
 
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import net.dv8tion.jda.api.events.interaction.ModalInteractionEvent;
 import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
-import net.dv8tion.jda.api.interactions.commands.OptionType;
+import net.dv8tion.jda.api.interactions.components.Modal;
+import net.dv8tion.jda.api.interactions.components.text.TextInput;
+import net.dv8tion.jda.api.interactions.components.text.TextInputStyle;
 
 import org.togetherjava.tjbot.features.CommandVisibility;
 import org.togetherjava.tjbot.features.SlashCommandAdapter;
 
+import java.time.Duration;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Optional;
+
 /**
  * The implemented command is {@code /chatgpt}, which allows users to ask ChatGPT a question, upon
  * which it will respond with an AI generated answer.
  */
 public final class ChatGptCommand extends SlashCommandAdapter {
-    private static final String QUESTION_OPTION = "question";
+    private static final String QUESTION_INPUT = "question";
+    private static final int MAX_MESSAGE_INPUT_LENGTH = 200;
+    private static final int MIN_MESSAGE_INPUT_LENGTH = 4;
+    private static final Duration COMMAND_COOLDOWN = Duration.of(10, ChronoUnit.SECONDS);
     private final ChatGptService chatGptService;
 
+    private final Cache<String, Instant> userIdToAskedAtCache =
+            Caffeine.newBuilder().maximumSize(1_000).expireAfterWrite(COMMAND_COOLDOWN).build();
+
     /**
      * Creates an instance of the chatgpt command.
-     * 
+     *
      * @param chatGptService ChatGptService - Needed to make calls to ChatGPT API
      */
     public ChatGptCommand(ChatGptService chatGptService) {
         super("chatgpt", "Ask the ChatGPT AI a question!", CommandVisibility.GUILD);
 
         this.chatGptService = chatGptService;
-
-        getData().addOption(OptionType.STRING, QUESTION_OPTION, "What do you want to ask?", true);
     }
 
     @Override
     public void onSlashCommand(SlashCommandInteractionEvent event) {
+        Instant previousAskTime = userIdToAskedAtCache.getIfPresent(event.getMember().getId());
+        if (previousAskTime != null) {
+            long timeRemainingUntilNextAsk = COMMAND_COOLDOWN.getSeconds()
+                    - Duration.between(previousAskTime, Instant.now()).get(ChronoUnit.SECONDS);
```

Comment:
```suggestion
            long timeRemainingUntilNextAsk = Duration.between(previousAskTime, Instant.now())
                .minus(COMMAND_COOLDOWN).get(ChronoUnit.SECONDS);
```
Styling is incorrect, but well, I'd personally do this.
Otherwise you're converting it twice to seconds, while you don't need to

Reply:
I have changed this, but what do you mean by the styling be incorrect?

Reply:
I was referring to spotless styling

Reply:
Unfortunately, the code provided gave value which counted down toward -10 instead of 0. I came up with this:

```
long timeRemainingUntilNextAsk = COMMAND_COOLDOWN
        .minus(Duration.between(previousAskTime, Instant.now()))
        .toSeconds();
```

 is it acceptable? I no longer convert to seconds twice.

Reply:
LOL, yea that code is correct. I swapped them around, my bad.

Once that's commited, looks good to me